gtfs/spec/en/reference.md

```diff
@@ -612,6 +623,69 @@ Describes levels in a station. Useful in conjunction with `pathways.txt`, and is
 |  `level_index` | Float | **Required** | Numeric index of the level that indicates its relative position. <br><br>Ground level should have index `0`, with levels above ground indicated by positive indices and levels below ground by negative indices.|
 |  `level_name` | Text | Optional | Name of the level as seen by the rider inside the building or station.<hr>_Example: Take the elevator to "Mezzanine" or "Platform" or "-1"._|
 
+### location_groups.txt
+
+File: **Optional**
+
+Primary key (`location_group_id`, `location_id`)
+
+Defines location groups that indicate groups of stops where a rider may request pickup or drop off.
+
+| Field Name | Type | Required | Description |
+| ---------- | ---- | ------------ | ----------- |
+| `location_group_id` | ID | **Required** | Identifies a location group. A location group is a group of stops that together indicate locations where a rider may request pickup or drop off.<br><br>Multiple entries in `location_groups.txt` can have the same `location_group_id`. | 
+| `stop_id` | Foreign ID referencing `stops.stop_id` | **Required** | Identifies a stop belonging to the location group. |
+| `location_group_name` | Text | Optional | Name of the location group. Must be defined either once, or exhaustively for a single `location_group_id`. |
+
+### locations.geojson
+
+File: **Optional**
+
+Defines GeoJSON locations that indicate groups of lat/lon coordinates defining zones where riders can request either pickup or drop off.
+- This file uses a subset of the GeoJSON format, described in [RFC 7946](https://tools.ietf.org/html/rfc7946).
+- The `locations.geojson` file must contain a `FeatureCollection`.
+- A `FeatureCollection` defines various stop locations where riders may request pickup or drop off.
+- Every GeoJSON `Feature` must have an `id`. The `id` belongs to the same namespace as `stop_id` in `stops.txt` and `location_group_id` in `location_groups.txt`.
+- Every GeoJSON `Feature` should have objects and associated keys according to the table below:
+
+|  Field Name | Type | Presence | Description |
+|  ------ | ------ | ------ | ------ |
+| -&nbsp;`type` | String | **Required** | `"FeatureCollection"` of locations. |
+| -&nbsp;`features` | Array | **Required** | Collection of `"Feature"` objects describing the locations. |
+| &nbsp;&nbsp;&nbsp;&nbsp;\-&nbsp;`type` | String | **Required** | `"Feature"` |
+| &nbsp;&nbsp;&nbsp;&nbsp;\-&nbsp;`id` | String | **Required** | Location ID belonging to the same namespace as `stops.stop_id`. |
```

Comment:
I'm still unsure how helpful these "namespace" statements actually are or if they just add more confusion. If it's decided to remove that language, maybe in it's place we can say some thing like "Primary Key for referencing GeoJSON locations of on-demand services." ...?

Would love to hear other folks' thoughts on the "namespace" language.

Reply:
I don't know what the original mean of "namespace" here, but now it seems more like a confusion than clarification to me.

I'd prefer to remove or alter this if we can't elaborate the meaning here.

Reply:
Perhaps one of the original drafters of the Flex v2 doc can shed some light.

Reply:
I think we should modify the language to say something like "the id for a stop, location and location_group must be used only once even these entity types".

Reply:
I'm straining to find some language that is both compact and unambiguous. Not sure if the above is clear enough.

Reply:
> I think we should modify the language to say something like "the id for a stop, location and location_group must be used only once even across these entity types".

Except isn't this an unnecessary restriction now that location `id` and `location_group_id` have their own foreign key column in stop_times.txt?

Reply:
In our implementation at least they are de-normalised into the same base type `StopLocation` and there are many implementations of them. I'm pretty sure that something would break if you gave a location and a stop the same ID.

One implementation's inner workings isn't a good argument for a spec requirement but to me it makes sense to make the ID of "things where a vehicle can stop" unique across all entity types so that you can be agnostic to the precise implementation in APIs.

Reply:
If this were not the case, then we would have to prefix all `StopLocation` ids with their type which is even more confusing to API consumers.

Reply:
That makes sense @leonardehrenfried, I am on board.

In that case, I believe this language is the clearest: "...must be unique across all `stop_id`, locations.GeoJSON `id`, and `location_group_id` values."

Reply:
Thanks @leonardehrenfried @westontrillium 
https://github.com/google/transit/pull/388/commits/7d10ffa97add72e49974cad95fbc620229885bcc